### ISSUE TITLE
Prefer const for literals that are assigned upon declaration and not re-assigned

### ISSUE THREAD

**Issue by Bartleby2718** [dotnet/roslyn#74943]
**Brief description:** My team prefers using `const`s for literals that are assigned upon declaration and not re-assigned. I'm sure there are other teams that'd agree, but I'm curious if there's an analyzer for this. If not, can we add one?

**Languages applicable:**

Is your analyzer C# only? VB Only? Or Both? - Both

**Code example that the analyzer should report:**
```cs
// before
private void PrintHelpText()
{
    var helpText = "This is how you do it."; // flag this variable because it's assigned upon declaration, not re-assigned
    Console.Write(helpText);
}
```
could be updated to
```cs
// after
private void PrintHelpText()
{
    const string helpText = "This is how you do it.";
    Console.Write(helpText);
}
```

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).
